Add "devpiserver_on_remove" hook for removing unpacked doczip files

drtuxwang opened the pull request:
Currently "devpi_web" unpacks "doc.zip" but these unpacked directories are never deleted when a ".doc.zip" file is removed from the devpi registry. This changes adds a new hook which calls into removal code in "devpi_web"

RonnyPfannschmidt commented:
looks good, can you add a test or do you need help with figuring the details

drtuxwang commented:
Looks like there is a "test_doczip_uploaded_hook" test in "server/test_devpi_server/test_model.py" but I cannot find any test relating to this in "web/devpi_web".

I can write a "test_doczip_remove_hook" test using this test as a guide. Unfortunately I am unable to work out how to run the tests.

drtuxwang commented:
Ronny,

Hello, I have added a test. But have problems working out how to run it.

Can you take look please?

Thanks

Colin

On 29 June 2017 at 16:40, Ronny Pfannschmidt <notifications@github.com>
wrote:

> looks good, can you add a test or do you need help with figuring the
> details
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/devpi/devpi/pull/438#issuecomment-312006075>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AMGTmHyxarchg7SB01M19W3L-_dipiKIks5sI8VTgaJpZM4OJc2f>
> .
>

fschulze commented:
I took a look at this. Compared to the other hooks, this new one is in the wrong place. The other hooks trigger from the event thread, the new one would trigger from the main thread. I tried moving it, but didn't get it working yet, because in the correct place I don't have enough data for a useful hook. I will investigate this further.

drtuxwang commented:
Hi, any news about merging this into master?

Thanks

Colin

On 14 August 2017 at 06:13, Florian Schulze <notifications@github.com>
wrote:

> I took a look at this. Compared to the other hooks, this new one is in the
> wrong place. The other hooks trigger from the event thread, the new one
> would trigger from the main thread. I tried moving it, but didn't get it
> working yet, because in the correct place I don't have enough data for a
> useful hook. I will investigate this further.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/devpi/devpi/pull/438#issuecomment-322101793>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AMGTmBcjNviyf_iXvf4E4akIXgdLleNrks5sX9digaJpZM4OJc2f>
> .
>

drtuxwang commented:
Hi I have successfully moved remove docs call to the TxNotificationThread and tested it. Can you take another look please? Thanks

fschulze commented on server/test_devpi_server/test_model.py:
```diff
@@ -780,6 +780,36 @@ def devpiserver_on_upload(self, stage, project, version, link):
+    @pytest.mark.start_threads
+    def test_doczip_remove_hook(self, stage, queue):
+        class Plugin:
+            def devpiserver_on_remove(self, stage, link):
```
The mock plugin needs to be updated for the new hook signature.

drtuxwang commented:
Hi, I have fixed the unit test and added a line to the ./server/CHANGELOG

However, I am unable to work out why tox has problems finding devpiserver_on_remove. Can you give me advice please? Thanks

fschulze commented:
I've fixed that issue and merged your changes after a rebase.

drtuxwang commented:
Thanks :)
